libnavigation-ui/src/main/java/com/mapbox/navigation/ui/internal/utils/BitmapEncodeOptions.kt

```diff
@@ -0,0 +1,18 @@
+package com.mapbox.navigation.ui.internal.utils
+
+import com.mapbox.navigation.ui.internal.utils.ViewUtils.DEFAULT_BITMAP_ENCODE_COMPRESS_QUALITY
+import com.mapbox.navigation.ui.internal.utils.ViewUtils.DEFAULT_BITMAP_ENCODE_WIDTH
+
+data class BitmapEncodeOptions(val width: Int, val compressQuality: Int) {
+    class Builder {
+        private var width: Int = DEFAULT_BITMAP_ENCODE_WIDTH
+        private var compressQuality: Int = DEFAULT_BITMAP_ENCODE_COMPRESS_QUALITY
+
+        fun width(width: Int) = apply { this.width = width }
+        fun compressQuality(compressQuality: Int) = apply { this.compressQuality = compressQuality }
+
+        fun build(): BitmapEncodeOptions {
+            return BitmapEncodeOptions(width, compressQuality)
+        }
+    }
+}
```

Comment:
👀 at this, I observed that this pattern has been replicated with what we are following for `Builder` pattern in `Core SDK`. Since, this is a data class, anyone can directly create an instance of it. Hence, it negates the need of `Builder` pattern. I think we should probably revisit this.
cc @JunDai @Guardiola31337 @LukasPaczos

Reply:
after having discussion with @Guardiola31337 and checking the discussion in this [Document recording design decisions](https://docs.google.com/document/d/16CpedUXG0yPxt0QPoWZrROA9UJVJ9Iwm6qCH2aEoS6c/edit?usp=sharing) which shared by @kmadsen 
I decided to go this way, use a class but not a data class, 👀 -> https://github.com/mapbox/mapbox-navigation-android/pull/2992#discussion_r426952125, and hide the constructor to force the developer using the Builder().

Given this is an internal package, I think it should be fine. But welcome team discussion around this design so our API can be consistent and compatible with potential change.

Ref: 
* https://github.com/mapbox/mapbox-navigation-android/pull/2992
* https://github.com/mapbox/mapbox-navigation-android/issues/2709
* https://github.com/mapbox/mapbox-navigation-android/pull/2624#discussion_r403145619

cc @LukasPaczos @Guardiola31337 @kmadsen @cafesilencio @abhishek1508

Reply:
Since this isn't a data class anymore, could you add the `equals/hashcode` override to `BitmapEncodeOptions`?

Reply:
we also preferred the ability to have a private constructor in the data class. but it doesn't matter with the kotlin copy constructor. 

how we were enforcing using a builder, was to require a Builder in the constructor. Passing the builder also gives you the ability to "rebuild" the object. for example:
``` kotlin
data class BitmapEncodeOptions(val width: Int, val compressQuality: Int, builder: Builder)
```
Also internal classes we have decided we don't need a builder in the constructor, Builder is helpful for populating default values and calling from Java code. So I'm also fine with:
``` kotlin
data class BitmapEncodeOptions(val width: Int, val compressQuality: Int) {
   class Builder() {
     // defaults
   }
}
```

Reply:
think it would be nice to keep the code consistent within Nav SDK. so I revert the change and to use the data class with Builder. @kmadsen - can you review it again?

Reply:
👍  thanks for addressing feedback! :shipit: